tests/cli-tests/compression/basic.sh

```diff
@@ -24,6 +24,10 @@ zstd -c file       | zstd -t
 zstd --stdout file | zstd -t
 println bob | zstd | zstd -t
 
+# Test preservation in gzip mode
+(exec -a gzip zstd -c file)       | zstd -t
```

Comment:
I believe `exec -a` is a `bash` thing, not a `sh` one.
We try to maintain `sh` compatibility to improve portability of the test suite.

Reply:
that appears to be the only way unless one of the methods listed here is more acceptable: https://unix.stackexchange.com/questions/250681/is-there-a-posix-way-of-setting-zeroth-argument-of-a-target-application

Reply:
@dirkmueller I just fixed this in PR #3055

Can you please add this test to a new test case in `zstd-symlinks/gzip.sh`?

Then the equivalent test case would be:

```
#!/bin/sh
set -e
ln -s $(which zstd) > ./gzip

./gzip -c file | zstd -t
./gzip --stdout file | zstd -t
```

You'd also need to add this line to `zstd-symlinks/setup`:

```
datagen > file
```

Reply:
Also, you should test that the file still exists explicitly, with `test -f file`.